ui/perfherder/compare/CompareTableView.jsx

```diff
@@ -59,7 +59,15 @@ export default class CompareTableView extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.location.search !== prevProps.location.search) {
+    const { validated } = this.props;
+
+    if (
+      this.props.location.search !== prevProps.location.search ||
```

Comment:
This check should be removed since it happens in the `withValidation` [HOC](https://github.com/mozilla/treeherder/blob/master/ui/perfherder/Validation.jsx#L53) (that's where the `validated` props come from).

Reply:
I would suggest to allow CompareTableView to decide when to make the request
since it is an operation that is time consuming. The component Validation does not know 
enough information to make this decision for CompareTableView.

Reply:
> The component Validation does not know enough information to make this decision for CompareTableView.

I don't follow what you mean. Is there a situation where the `validated` props are not updated in `withValidation` when the relevant query strings for the compare view are changed? 

To be clear, I'm not saying that we should not be checking for prop changes in the `CompareTableView` to update performance data. What I'm saying is that it's redundant to check for location props here since it's done in `withValidation` and those updated props - after they are checked for errors - are passed to the `CompareTableView`.

Reply:
> > The component Validation does not know enough information to make this decision for CompareTableView.
> 
> I don't follow what you mean. Is there a situation where the `validated` props are not updated in `withValidation` when the relevant query strings for the compare view are changed?

Yes, the `validated` props aren't updated on time. While debugging, this is the flow we discovered:
1. We've changed the `newRevision` param.
2. `CompareTableView.componentDidUpdate()` kicks in & reaches `this.props.location.search !== prevProps.location.search`, which evaluates to true *initialy*. But the `validated` props aren't up to date *yet*. Thus, `CompareTableView` made a redundant backend request.
3. Then `validated` props get updated, causing `CompareTableView.componentDidUpdate()` to kick in again. *But*, `this.props.location.search !== prevProps.location.search` no longer evaluates to true; `location` from new props & previous props are now the same.